src/azul/drs.py

```diff
@@ -185,6 +177,42 @@ def drs_object(object_uuid, version=None):
     }
 
 
+class AccessMethod(Enum):
+    https: str = 'https'
+    gs: str = 'gs'
+
+
+@dataclass
+class DRSObject:
+    """"
+    Used to build up a https://ga4gh.github.io/data-repository-service-schemas/docs/#_drsobject
+    """
+    uuid: str
+    version: Optional[str] = None
+    access_methods: List[MutableJSON] = field(default_factory=list)
+
+    def add_access_method(self,
+                          access_method: AccessMethod, *,
+                          url: Optional[str] = None,
+                          access_id: Optional[str] = None):
+        """
+        We only currently use `url_type`s of 'https' and 'gs'. Only one of `url`
+        and `access_id` should be specified.
+        """
+        assert url is None or access_id is None
+        self.access_methods.append({
+            **({'access_url': {'url': url}} if url else {}),
+            **({'access_id': access_id} if access_id else {}),
+            'type': access_method.value
+        })
+
+    def serialize(self) -> JSON:
```

Comment:
```suggestion
    def to_json(self) -> MutableJSON:
```

Reply:
Why MutableJSON? Changes to the JSON could unintentionally change the original object.

Reply:
@hannes-ucsc: "We should restrict to JSON."